workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/WorkflowNode.kt

```diff
@@ -209,16 +214,27 @@ internal class WorkflowNode<PropsT, StateT, OutputT, RenderingT>(
    *    time of suspending.
    */
   @OptIn(ExperimentalCoroutinesApi::class, DelicateCoroutinesApi::class)
-  fun onNextAction(selector: SelectBuilder<ActionProcessingResult>): Boolean {
-    // Listen for any child workflow updates.
-    var empty = subtreeManager.onNextChildAction(selector)
+  fun onNextAction(
+    selector: SelectBuilder<ActionProcessingResult>,
+    skipChangedNodes: Boolean = false
+  ): Boolean {
+    var empty = if (!skipChangedNodes || !selfStateDirty) {
+      // Listen for any child workflow events.
+      subtreeManager.onNextChildAction(selector, skipChangedNodes)
+    } else {
+      // Our state changed and we are skipping changed nodes, so our actions are empty from
+      // this node down.
+      true
+    }
 
     empty = empty && (eventActionsChannel.isEmpty || eventActionsChannel.isClosedForReceive)
 
-    // Listen for any events.
-    with(selector) {
-      eventActionsChannel.onReceive { action ->
-        return@onReceive applyAction(action)
+    if (!skipChangedNodes || !selfStateDirty) {
```

Comment:
Nit: Can either of these properties change between line 221 and here? If not, consider assignin to a local val to make it more clear that both these ifs use the same condition.

Reply:
no they can't. and sure I can do that!